### ISSUE TITLE
Admins should be able to add a new source when editing documents

### ISSUE THREAD

**Issue by jameslaneconkling** [crowdcover/congo-mines-app#71]
Currently, it's only possible to select source from a dropdown of existing sources.  It should also be possible to input a new source.

**Comment by markmcdonald51** [comment]
Hey James, I can add this in but is there any real data in this table in the source field? Going to wait till you get back to me to do this operation.

**Comment by markmcdonald51** [comment]
done.

**Comment by jameslaneconkling** [comment]
> Hey James, I can add this in but is there any real data in this table in the source field? Going to wait till you get back to me to do this operation.

@markmcdonald51  Yeah, there's all of [these sources](http://188.166.93.154/admin/sources) that were created in the scrape.  This is good, as it means that the same source won't have variants in spelling.  But if they want to  create a report from a source that hasn't yet been defined in the source table, there's no way to do this, since the source form isn't exposed to them yet.

**Comment by jameslaneconkling** [comment]
@markmcdonald51 wait, unless I'm mistaken, it looks like you added this to deposits, not documents.  In short:

* the source field for **deposits** should be free text (no association, and no need for the table `deposit sources`)
* the source field for **documents** is already an association with the [source table](http://188.166.93.154/admin/sources).  This is good, but when adding new documents, admins can only choose from existing sources, they can't create a new source if necessary.  See below screenshot.

![screenshot 2015-06-03 18 35 44](https://cloud.githubusercontent.com/assets/4389360/7975145/e23a66e4-0a1f-11e5-944d-37e117e8913f.png)
